### ISSUE TITLE
ALB limits

### ISSUE THREAD

**Issue by drew222** [jantman/awslimitchecker#268]
My understanding is that Amazon limits load balancers based on the sum of classic and application load balancers. This library seems to get the number of classic 'Active load balancers' on the 'ELB' service. Is it possible to sum together the classic and application load balancers values when checking against the 'Active load balancers' limit? 

should be limit_type='AWS::ElasticLoadBalancingV2::LoadBalancer'

**Comment by jantman** [comment]
@drew222 According to the [service limits documentation](http://docs.aws.amazon.com/general/latest/gr/aws_service_limits.html#limits_elastic_load_balancer), there are separate limits for ELBs and ALBs.

awslimitchecker currently only supports Classic load balancers, not ALBs.

As of today, there is a note on the [Trusted Advisor FAQs page](https://aws.amazon.com/premiumsupport/ta-faqs/) that says:

> Note: The **Active Load Balancers** limit check is currently inaccurate. Although the limit applies to the total number of load balancers, the check counts only Classic Load Balancers and does not count Application Load Balancers.

As such, I'd prefer to wait on adding ALB support, since there is currently no way to programmatically retrieve an account's ALB limit; it's not available via any API, Trusted Advisor, or the [EC2 Service Limits Report](https://aws.amazon.com/about-aws/whats-new/2014/06/19/amazon-ec2-service-limits-report-now-available/) (the "Limits" tab on the EC2 Console). So we don't even know what AWS will call these limits, and there's no way for awslimitchecker to retrieve the current value short of each user tracking their limit increase requests and manually updating via a [limit override](http://awslimitchecker.readthedocs.io/en/latest/cli_usage.html#overriding-limits).

For what it's worth, my employer has an Enterprise-level account, and we're trying to use that to push for an actual API call that exposes all EC2-related current limits.

**Comment by drew222** [comment]
Ahh okay, sounds like a waiting game. Thanks for the timely and detailed response!

**Comment by tamsky** [comment]
> there is currently no way to programmatically retrieve an account's ALB limit; it's not available via any API, Trusted Advisor, or the ... ("Limits" tab on the EC2 Console).

Has this situation changed?

My Limits tab [1] shows ELB, ALB, and Target Group limits:

![image](https://cloud.githubusercontent.com/assets/96978/26756459/ad45bab0-4857-11e7-93e2-24720c407a7a.png)

Which appears to derive from two different POST calls.

The ALB response data from https://console.aws.amazon.com/ec2/ecb?call=elbV2DescribeLoadBalancerAccountLimits?  is:

``` json
{
  "limits": [
    {
      "name": "application-load-balancers",
      "max": "500"
    },
    {
      "name": "target-groups",
      "max": "400"
    },
    {
      "name": "targets-per-application-load-balancer",
      "max": "1000"
    },
    {
      "name": "listeners-per-application-load-balancer",
      "max": "10"
    },
    {
      "name": "rules-per-application-load-balancer",
      "max": "100"
    }
  ]
}
```

The ELB limits come from https://console.aws.amazon.com/ec2/ecb?call=describeLoadBalancerAccountLimits? which returns
`{"limits":[{"max":"500","name":"classic-load-balancers"},{"max":"100","name":"classic-listeners"}]}`

[1] https://console.aws.amazon.com/ec2/v2/home?region=us-east-1#Limits:

**Comment by jantman** [comment]
Yes, it does appear the situation has changed. There are now [ELB DescribeAccountLimits](http://docs.aws.amazon.com/elasticloadbalancing/2012-06-01/APIReference/API_DescribeAccountLimits.html) and [ELBv2 DescribeAccountLimits](http://docs.aws.amazon.com/elasticloadbalancing/latest/APIReference/API_DescribeAccountLimits.html) API calls. 

Unless something is broken, right now the [EC2 Release Notes](https://aws.amazon.com/releasenotes/Amazon-EC2) page doesn't show anything since 2014. So, as has been the case for years, nobody knows when new features are released. My employer has an Enterprise account and lack of API access to limits is one of the issues we've filed with our TAMs, and _we_ haven't been told about this yet... According to the [git history](https://github.com/boto/botocore/commits/develop/botocore/data/elb/2012-06-01/service-2.json) of the ELB API in botocore, the official Python API client (and library that awslimitchecker uses), the DescribeAccountLimits calls were [added](https://github.com/boto/botocore/commit/b7b4d715dbf8e3d79802fd6895300a33bb85744b) sometime between April 6 and May 11 of this year.

Thanks for bringing this to my attention, @tamsky. I'll update the labels on this issue and try to start work on it sometime this week.

**Comment by jantman** [comment]
This has been released in [0.11.0](https://github.com/jantman/awslimitchecker/releases/tag/0.11.0)
